torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -203,10 +203,12 @@ DebugInfoWriter& DebugInfoWriter::getWriter(int rank) {
 }
 
 void DebugInfoWriter::registerWriter(std::unique_ptr<DebugInfoWriter> writer) {
-  TORCH_CHECK_WITH(
-      DistBackendError,
-      hasWriterRegistered_.load() == false,
-      "debugInfoWriter already registered");
+  if (hasWriterRegistered_.load()) {
+    TORCH_WARN_ONCE(
+      "DebugInfoWriter is registered more than once, but since we need it to stay "
+      "outside ProcessGroup, user needs to ensure that this registration is indeed needed. "
+      "And we will only use the last registered writer.");
```

Comment:
How about:
"DebugInfoWriter has already been registered, ignoring"

Reply:
Well, the logic here is let the later registration overriding the old one. lol.